gum/backend-arm64/gumstalker-arm64.c

```diff
@@ -439,6 +441,64 @@ gum_stalker_finalize (GObject * object)
   G_OBJECT_CLASS (gum_stalker_parent_class)->finalize (object);
 }
 
+static gboolean
+gum_stalker_clone_checker (GumCpuContext *cpu,
+                           GumExecCtx *ctx)
+{
+#ifdef HAVE_LINUX
+  guint64 svc_number = cpu->x[8];
+
+  if (svc_number == __NR_clone)
+    return TRUE;
+
+  return FALSE;
+#else
+  return FALSE;
+#endif
+}
+
+static GumVirtualizationRequirements
+gum_exec_block_virtualize_linux_sysenter_insn (GumExecBlock * block,
+                                               GumGeneratorContext * gc)
+{
+  GumArm64Writer *cw = gc->code_writer;
+  gconstpointer continue_normally = cw->code + 1;
+  gconstpointer end_payload = cw->code + 3;
+  const cs_insn * insn = gc->instruction->ci;
+
+  gum_arm64_relocator_skip_one (gc->relocator);
+
+  if (gc->opened_prolog != GUM_PROLOG_NONE)
+    gum_exec_block_close_prolog (block, gc);
+
+  gum_exec_block_open_prolog (block, GUM_PROLOG_FULL, gc);
+
+  gum_arm64_writer_put_call_address_with_arguments (gc->code_writer,
+      GUM_ADDRESS (gum_stalker_clone_checker), 2,
+      GUM_ARG_REGISTER, ARM64_REG_X20,
+      GUM_ARG_ADDRESS, GUM_ADDRESS (block->ctx));
```

Comment:
How about just saving the flags, doing the compare, and then restoring them in each branch? A full prolog/epilog with helper function here seems like a lot of unnecessary overhead.